plugins/flatpak/gs-flatpak.c

```diff
@@ -3703,14 +3702,20 @@ gs_flatpak_has_space_to_install (GsFlatpak *self, GsApp *app, GsAppList *list)
 {
 	g_autoptr(GError) error = NULL;
 	guint64 free_space = 0;
+	guint64 min_free_space = 0;
 	guint64 space_required = 0;
 
 	space_required = gs_app_get_size_download (app);
 	if (space_required == GS_APP_SIZE_UNKNOWABLE) {
 		g_warning ("Failed to query download size: %s", gs_app_get_unique_id (app));
 		space_required = 0;
 	}
-	space_required = space_required + EOS_MIN_FREE_SPACE_NEEDED_BYTES;
+	if (!flatpak_installation_get_min_free_space_bytes (self->installation, &min_free_space, &error)) {
+		g_autoptr(GFile) installation_file = flatpak_installation_get_path (self->installation);
+		g_autofree gchar *path = g_file_get_path (installation_file);
+		g_warning ("Error getting min-free-space config value of OSTree repo at %s:%s", path, error->message);
```

Comment:
You need to call `g_clear_error (&error)` before leaving this branch, otherwise a non-`NULL` error will be passed in to `get_installation_dir_free_space()`, which it’s likely not prepared for (so will cause a leak or crash).